src/bms/lex/token.rs

```diff
@@ -373,6 +394,184 @@ impl<'a> Token<'a> {
+                    }
+                    let alpha = parts[0]
+                        .parse()
+                        .map_err(|_| c.make_err_expected_token("invalid alpha value"))?;
+                    let red = parts[1]
+                        .parse()
+                        .map_err(|_| c.make_err_expected_token("invalid red value"))?;
+                    let green = parts[2]
+                        .parse()
+                        .map_err(|_| c.make_err_expected_token("invalid green value"))?;
+                    let blue = parts[3]
+                        .parse()
+                        .map_err(|_| c.make_err_expected_token("invalid blue value"))?;
+
+                    Self::ExBmp(
+                        ObjId::from(id, c)?,
+                        Argb {
+                            alpha,
+                            red,
+                            green,
+                            blue,
+                        },
+                        Path::new(filename),
+                    )
+                }
+                exrank if exrank.starts_with("#EXRANK") => {
+                    let id = exrank.trim_start_matches("#EXRANK");
+                    let level = c
+                        .next_token()
+                        .ok_or_else(|| c.make_err_expected_token("judge level"))?;
+
+                    let judge_level = match level {
+                        "0" => JudgeLevel::VeryHard,
+                        "1" => JudgeLevel::Hard,
+                        "2" => JudgeLevel::Normal,
+                        "3" => JudgeLevel::Easy,
+                        "4" => JudgeLevel::VeryEasy,
+                        _ => {
+                            return Err(
+                                c.make_err_expected_token("expected one of 0, 1, 2, 3 or 4")
+                            );
+                        }
+                    };
+
+                    Self::ExRank(ObjId::from(id, c)?, judge_level)
+                }
+                exwav if exwav.starts_with("#EXWAV") => {
+                    let id = exwav.trim_start_matches("#EXWAV");
+                    let p1 = c
+                        .next_token()
+                        .ok_or_else(|| c.make_err_expected_token("param1"))?;
+                    let p2 = c
+                        .next_token()
+                        .ok_or_else(|| c.make_err_expected_token("param2"))?;
+                    let p3 = c
+                        .next_token()
+                        .ok_or_else(|| c.make_err_expected_token("param3"))?;
+                    let filename = c
+                        .next_token()
+                        .ok_or_else(|| c.make_err_expected_token("filename"))?;
+                    Self::ExWav(
+                        ObjId::from(id, c)?,
+                        [p1, p2, p3, filename],
+                        Path::new(filename),
+                    )
```

Comment:
According to [BMS command memo](https://hitkey.nekokan.dyndns.info/cmdsJP.htm#EXWAVXX), number of `#EXWAV`'s parameters varies by the first parameter flags. So this parsing will fail when the first one has less than 2 items.